src/execution/StepExecutionManager.ts

```diff
@@ -62,6 +70,66 @@ export class StepExecutionManager {
     let process = statusManager.findOrCreateProcess(step, currentProcessType)
 
     if (process.status !== 'DONE') {
+      const multisigProcess = step.execution.process.find(
+        (p) => !!p.multisigInternalTxHash
+      )
+
+      // TODO: update @lifi/types repo for these changes [DO NOT MERGE IF YOU SEE THIS]
+      if (isMultisigSigner && multisigProcess) {
+        if (
+          !multisigProcess ||
+          !config.multisigConfig?.getMultisigTransactionDetails
+        ) {
+          throw new Error('MultiSig process is undefined')
+        }
+        if (!config.multisigConfig?.getMultisigTransactionDetails) {
+          throw new Error(
+            '"getMultiSigTransactionDetails()" is missing in MultiSig config.'
+          )
+        }
+
+        const multisigInternalTxHash = multisigProcess.multisigInternalTxHash
+
+        if (!multisigInternalTxHash) {
+          // need to check what happens in failed tx
+          throw new Error('Multisig internal transaction hash is undefined.')
+        }
+
+        const response: MultiSigTxDetails =
+          await config.multisigConfig?.getMultisigTransactionDetails(
+            multisigInternalTxHash
+          )
+
+        if (response.status === 'PENDING') {
+          return step.execution!
+        }
+
+        if (response.status === 'SUCCESS') {
+          statusManager.updateExecution(step, 'DONE')
+          process = statusManager.updateProcess(step, process.type, 'DONE', {
+            txHash: response.txHash,
+            txLink:
+              fromChain.metamask.blockExplorerUrls[0] + 'tx/' + response.txHash,
+          })
+
+          return step.execution!
```

Comment:
Are you sure this can be `DONE` on this stage? 👀

Reply:
yes, this is when the status from SAFE Interaction is completed

Reply:
> yes, this is when the status from SAFE Interaction is completed

I think not because in that case we say that the execution of the whole step is DONE, which is not true at this stage.

I assume we have SUCCESS status for multisig transaction, which means all required signers have signed the transaction and we now should wait for it to be mined. Am I missing something here?

Reply:
Kinda yes, afaik the SAFE status would be success only when the transaction in mined and all the other execution process is done.